src/graphics/image_factory.cpp

```
@@ -102,8 +102,18 @@ namespace moth_graphics::graphics {
             m_cachedImages.insert(std::make_pair(key, cacheDesc));
             return m_context.NewImage(texture, sourceRect);
         }
+        if (m_fallbackDesc) {
+            spdlog::warn("ImageFactory: '{}' not found, using fallback", path.string());
+            return m_context.NewImage(m_fallbackDesc->m_texture, m_fallbackDesc->m_sourceRect);
+        }
+        spdlog::error("ImageFactory: '{}' not found and no fallback set", path.string());
         return nullptr;
     }
+
+    void ImageFactory::SetFallbackImage(std::shared_ptr<ITexture> texture) {
```

Clang (14.0.6)

[error] variable 'SetFallbackImage' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors)